cocotb_test/simulator.py

```diff
@@ -256,6 +265,17 @@ def outdated(self, output, dependencies):
 
         return False
 
+    def exit_gracefully(self, signum, frame):
+        pid = None
+        if self.process is not None:
+            pid = self.process.pid
+            self.process.stdout.flush()
+            self.process.kill()
+            self.process.wait()
+        # Restore previous handlers
+        signal.signal(signal.SIGINT, self.old_sigint_h)
+        signal.signal(signal.SIGTERM, self.old_sigterm_h)
+        pytest.fail("Exiting pid: " + str(pid) + " with signum: " + str(signum), False)
```

Comment:
Can we simply assert here? It would be potentially better no to depend on `pytest` here.

Like here: 
https://github.com/themperek/cocotb-test/blob/58cd9be1c0530b4982b45c214e338a66483e31e3/cocotb_test/simulator.py#L189

Reply:
Yes, no probem